projects/core/src/lib/operators/dialogs/column-range-filter/column-range-filter.component.ts

```diff
@@ -72,54 +106,131 @@ export class ColumnRangeFilterComponent implements OnInit, OnDestroy {
+            },
+        );
+    }
+
+    processVectorLayer(_layer: VectorLayer, metadata: VectorLayerMetadata, data: VectorData): void {
+        if (!data.data || (data.data && data.data.length === 0)) {
+            return;
+        }
+        this.dataStream = data;
+    }
 
     //control over filters
-    get filters(): UntypedFormArray {
-        return this.form.get('filters') as UntypedFormArray;
+    get filters(): FormArray<FormGroup<FilterForm>> {
+        return this.form.get('filters') as FormArray<FormGroup<FilterForm>>;
     }
 
-    newFilter(): UntypedFormGroup {
+    newFilter(): FormGroup<FilterForm> {
         return this.formBuilder.group({
-            attribute: '',
-            ranges: this.formBuilder.array([]),
+            attribute: this.formBuilder.nonNullable.control<string>(''),
+            ranges: this.formBuilder.array<FormGroup<RangeForm>>([]),
         });
     }
 
     addFilter(): void {
         this.filters.push(this.newFilter());
-        this.addRange(this.filters.length - 1);
+        this.addRange(this.filters.length - 1, '', '');
     }
 
     removeFilter(i: number): void {
+        //remove histogram
+        this.removeHistogram(i);
         this.filters.removeAt(i);
     }
 
     //control over the ranges
-    ranges(filterIndex: number): UntypedFormArray {
-        return this.filters.at(filterIndex).get('ranges') as UntypedFormArray;
+    ranges(filterIndex: number): FormArray<FormGroup<RangeForm>> {
+        return this.filters.at(filterIndex).get('ranges') as FormArray<FormGroup<RangeForm>>;
     }
 
-    newRange(): UntypedFormGroup {
+    newRange(min: string, max: string): FormGroup<RangeForm> {
         return this.formBuilder.group({
-            min: '',
-            max: '',
+            min: this.formBuilder.nonNullable.control<string>(min),
+            max: this.formBuilder.nonNullable.control<string>(max),
         });
     }
 
-    addRange(filterIndex: number): void {
-        this.ranges(filterIndex).push(this.newRange());
+    addRange(filterIndex: number, min: string, max: string): void {
+        this.ranges(filterIndex).push(this.newRange(min, max));
     }
 
+    removeAllRanges(filterIndex: number): void {
+        this.ranges(filterIndex).clear();
+    }
     removeRange(filterIndex: number, rangeIndex: number): void {
         this.ranges(filterIndex).removeAt(rangeIndex);
     }
 
+    updateRange(filterIndex: number, rangeIndex: number, min: number, max: number): void {
+        this.ranges(filterIndex).at(rangeIndex).setValue({min: min.toString(), max: max.toString()});
+    }
+
+    resetFiltersAndRanges(): void {
+        for (let index = 0; index < this.filters.controls.length; index++) {
+            this.removeAllRanges(index);
+        }
+        this.filters.reset();
+        this.histograms = new Map<number, ReplaySubject<VegaChartData>>();
+        if (this.histogramSubscription) {
+            this.histogramSubscription.unsubscribe();
+        }
```

Comment:
Is there only one subscription?
What about having two histograms for two attributes?

But they somehow both seem to update :thinking: .

Reply:
d07fd67
As the histogram data is subscribed with the async pipe in the template, it will unsubscribe automatically. So I removed the histogramSubscription. That's why both updated ;-)